src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_DeconstructionAssignmentOperator.cs

```diff
@@ -31,18 +31,25 @@ public override BoundNode VisitDeconstructionAssignmentOperator(BoundDeconstruct
         /// - the conversion phase
         /// - the assignment phase
         /// </summary>
-        private BoundExpression RewriteDeconstruction(BoundTupleExpression left, Conversion conversion, BoundExpression right)
+        private BoundExpression RewriteDeconstruction(BoundTupleExpression left, Conversion conversion, BoundExpression right, bool isUsed)
         {
             var temps = ArrayBuilder<LocalSymbol>.GetInstance();
             var effects = DeconstructionSideEffects.GetInstance();
             ArrayBuilder<Binder.DeconstructionVariable> lhsTargets = GetAssignmentTargetsAndSideEffects(left, temps, effects.init);
 
-            BoundExpression returnTuple = ApplyDeconstructionConversion(lhsTargets, right, conversion, temps, effects, inInit: true);
-            if (!returnTuple.HasErrors)
+            BoundExpression returnValue = ApplyDeconstructionConversion(lhsTargets, right, conversion, temps, effects, inInit: true);
+            if (!returnValue.HasErrors)
             {
-                returnTuple = VisitExpression(returnTuple);
+                if (isUsed)
+                {
+                    returnValue = VisitExpression(returnValue);
+                }
+                else
+                {
+                    returnValue = new BoundUnusedResult(left.Syntax, left.Type);
```

Comment:
>left.Type [](start = 69, length = 9)

Is this possibly an error type? This doesn't feel right, we probably shouldn't have any references to error types after lowering.